README.md
![Fluorine](Assets/logo.png)

Fluorine is part of the **[E-sites iOS Suite](https://github.com/e-sites/iOS-Suite)**.

---

A small UIView helper to create simple linear gradients.

[![forthebadge](http://forthebadge.com/images/badges/made-with-swift.svg)](http://forthebadge.com) [![forthebadge](http://forthebadge.com/images/badges/built-with-swag.svg)](http://forthebadge.com)

[![Platform](https://img.shields.io/cocoapods/p/Fluorine.svg?style=flat)](http://cocoadocs.org/docsets/Fluorine)
[![CocoaPods Compatible](https://img.shields.io/cocoapods/v/Fluorine.svg)](http://cocoadocs.org/docsets/Fluorine)
[![Quality](https://apps.e-sites.nl/cocoapodsquality/Fluorine/badge.svg?004)](https://cocoapods.org/pods/Fluorine/quality)
[![Travis-ci](https://travis-ci.org/e-sites/Fluorine.svg?branch=master&001)](https://travis-ci.org/e-sites/Fluorine)


# Installation

Podfile:

```ruby
pod 'Fluorine'
```

And then

```
pod install
```

# Implementation

The following code will make 4 gradient views:    
![Screenshot](Assets/screenshot.png)

```swift
let view1 = Fluorine.View(frame: CGRect(x: 50, y: 50, width: 100, height: 100))
self.view.addSubview(view1)

let view2 = Fluorine.View(frame: CGRect(x: 160, y: 50, width: 100, height: 100))
view2.colors = [ UIColor.red, UIColor.green, UIColor.blue ]
self.view.addSubview(view2)

let view3 = Fluorine.View(frame: CGRect(x: 270, y: 50, width: 100, height: 100))
view3.colors = [ UIColor.orange, UIColor.black ]
view3.point = Fluorine.Point(start: CGPoint.zero, end: CGPoint(x: 1, y: 1))
self.view.addSubview(view3)

let view4 = Fluorine.View(frame: CGRect(x: 380, y: 50, width: 100, height: 100))
view4.colors = [ UIColor.red, UIColor.blue ]
view4.locations = [ 0.5, 1 ]
self.view.addSubview(view4)
```

CHANGELOG.md
# Changelog Fluorine

## v3.1.1 (18-19-2019)
- Swift PM

## v3.1.0 (05-04-2019)
- Swift 5.0

## v3.0.1 (16-07-2018)
- `Fluorine` class removed

## v3.0 (12-07-2018)
- Initial release

Sources/Fluorine.swift
//  Fluorine.swift
//  Library
//
//  Created by Bas van Kuijck on 13-01-16.
//  Copyright © 2016 E-sites. All rights reserved.
//
import Foundation
import UIKit

public struct Point {
    public let start: CGPoint
    public let end: CGPoint

    public init(start: CGPoint = CGPoint.zero, end: CGPoint = CGPoint(x: 0, y: 1)) {
        self.start = start
        self.end = end
    }
}

open class View: UIView {
    open var colors: [UIColor] = [ UIColor.black, UIColor(white: 0, alpha: 0) ] {
        didSet {
            _redraw()
        }
    }

    open var point: Point = Point() {
        didSet {
            _redraw()
        }
    }

    open var locations: [CGFloat] = [0, 1] {
        didSet {
            _redraw()
        }
    }

    // MARK: - Constructor
    // ____________________________________________________________________________________________________________________

    public init() {
        super.init(frame: CGRect.zero)
        _init()
    }

    public override init(frame aFrame: CGRect) {
        super.init(frame: aFrame)
        _init()
    }

    required public init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        _init()
    }

    private func _init() {
        self.isOpaque = false
        _redraw()

    }

    override open class var layerClass: AnyClass {
        return CAGradientLayer.self
    }

    // MARK: - Drawing
    // ____________________________________________________________________________________________________________________

    private func _redraw() {
        let layer = self.layer as! CAGradientLayer
        layer.colors = colors.map { $0.cgColor }
        layer.startPoint = point.start
        layer.endPoint = point.end
        layer.locations = locations as [NSNumber]?
        self.backgroundColor = UIColor.clear
    }
}



Project/Project/AppDelegate.swift
//
//  AppDelegate.swift
//  Project
//
//  Created by Bas van Kuijck on 03/11/2017.
//  Copyright © 2017 E-sites. All rights reserved.
//

import UIKit

@UIApplicationMain
class AppDelegate: UIResponder, UIApplicationDelegate {

    var window: UIWindow?

    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
        // Override point for customization after application launch.
        return true
    }

    func applicationWillResignActive(_ application: UIApplication) {
        // Sent when the application is about to move from active to inactive state. This can occur for certain types of temporary interruptions (such as an incoming phone call or SMS message) or when the user quits the application and it begins the transition to the background state.
        // Use this method to pause ongoing tasks, disable timers, and invalidate graphics rendering callbacks. Games should use this method to pause the game.
    }

    func applicationDidEnterBackground(_ application: UIApplication) {
        // Use this method to release shared resources, save user data, invalidate timers, and store enough application state information to restore your application to its current state in case it is terminated later.
        // If your application supports background execution, this method is called instead of applicationWillTerminate: when the user quits.
    }

    func applicationWillEnterForeground(_ application: UIApplication) {
        // Called as part of the transition from the background to the active state; here you can undo many of the changes made on entering the background.
    }

    func applicationDidBecomeActive(_ application: UIApplication) {
        // Restart any tasks that were paused (or not yet started) while the application was inactive. If the application was previously in the background, optionally refresh the user interface.
    }

    func applicationWillTerminate(_ application: UIApplication) {
        // Called when the application is about to terminate. Save data if appropriate. See also applicationDidEnterBackground:.
    }


}



Project/Project/ViewController.swift
//
//  ViewController.swift
//  Project
//
//  Created by Bas van Kuijck on 03/11/2017.
//  Copyright © 2017 E-sites. All rights reserved.
//

import UIKit
import Fluorine

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        let view1 = Fluorine.View(frame: CGRect(x: 50, y: 50, width: 100, height: 100))
        self.view.addSubview(view1)

        let view2 = Fluorine.View(frame: CGRect(x: 160, y: 50, width: 100, height: 100))
        view2.colors = [ UIColor.red, UIColor.green, UIColor.blue ]
        self.view.addSubview(view2)

        let view3 = Fluorine.View(frame: CGRect(x: 270, y: 50, width: 100, height: 100))
        view3.colors = [ UIColor.orange, UIColor.black ]
        view3.point = Fluorine.Point(start: CGPoint.zero, end: CGPoint(x: 1, y: 1))
        self.view.addSubview(view3)

        let view4 = Fluorine.View(frame: CGRect(x: 380, y: 50, width: 100, height: 100))
        view4.colors = [ UIColor.red, UIColor.blue ]
        view4.locations = [ 0.5, 1 ]
        self.view.addSubview(view4)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }


}

